### PR TITLE
Added package io.github.axisangles.ktmath

### PR DESCRIPTION
**Pull request by AxisAngles**
Added new kotlin math library, io.github.axisangles.ktmath, to replace jME3.math
includes Quaternion, Matrix3, Vector3, and EulerAngles.

- Contains mathematically propper operator overloads for matrices, vectors and quaternions.
- Has a small focus on numerical stability, trying to guarantee at least 6 sig figs on most operations.
- Contains several improvements over standard algorithms for converting to eulerangles.
- Provides 6 orders of eulerangles.
- Provides for homomorphic conversion from quaternion to matrix.
- Provides for matrix orthonormalization.

### CODE REVIEW

**Comment by Eirenliel** [comment]
Why not include as a lib or a fork?

**Comment by TheDevMinerTV** [comment]
(through gradle)

**Comment by ImUrX** [comment]
We discussed it and it's worth keeping it in here XD. Gonna start reading :P

File: server/src/main/java/io/github/axisangles/ktmath/Vector3.kt [hunk 1]
```diff
@@ -0,0 +1,92 @@
+@file:Suppress("unused")
+
+package io.github.axisangles.ktmath
+
+import kotlin.math.atan2
+import kotlin.math.sqrt
+
+data class Vector3(val x: Float, val y: Float, val z: Float) {
```
Comment (ButterscotchV) [inline server/src/main/java/io/github/axisangles/ktmath/Vector3.kt:8]: No "distance" method? It's pretty standard for most implementations to have one.

**Review by ButterscotchV — APPROVED** [review]
Well done on the unit tests, I'm really happy with how this looks.

File: server/src/main/java/io/github/axisangles/ktmath/Vector3.kt (hunk 1 above)
Comment (AxisAngles) [reply server/src/main/java/io/github/axisangles/ktmath/Vector3.kt:8]: There might be some methods missing. I felt (b - a).len() was sufficient but maybe not. Same goes for some methods in the matrix library like angleTo, and angle. if you think of any features you want to see, let me know and I'll add them shortly.

Comment (ButterscotchV) [reply server/src/main/java/io/github/axisangles/ktmath/Vector3.kt:8]: `(b - a).len()` is definitely sufficient, I just figure shortcuts like `.distance(Vector3 other)` are good to have as it'll be used really frequently

Comment (ImUrX) [reply server/src/main/java/io/github/axisangles/ktmath/Vector3.kt:8]: I personally coming from a math side of things prefer ``(b - a).len()`` for representing distance. They are as short and they are more explicit.

Comment (AxisAngles) [reply server/src/main/java/io/github/axisangles/ktmath/Vector3.kt:8]: I think it's probably worth it though I suppose someone or I can always add these additions in the future.

**Review by Eirenliel — APPROVED** [review]
Besides the fact that unit tests are not done properly (usually you need to check all edge cases) :D
